### PR TITLE
Add Fisher-Yates Shuffle Algorithm

### PR DESCRIPTION
**Pull request by Dima1247**
Added Fisher-Yates Shuffle algorithm.
Added tests based on implementation in JavaScript repo.

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

**Comment by Dima1247** [comment]
Hi, created this PR just to provide an implementation of Fisher-Yates Shuffle on .NET.
Any  suggestions are welcome 😉

**Comment by Dima1247** [comment]
I'm not sure about directory in which I placed this algorithm. I can change it if it's wrong.

File: Algorithms.Tests/Sorters/Comparison/FisherYatesShufflerTests.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+using Algorithms.Sorters.Comparison;
+using Algorithms.Tests.Helpers;
+using NUnit.Framework;
+
+namespace Algorithms.Tests.Sorters.Comparison
+{
+    public static class FisherYatesShufflerTests
+    {
+        [Test]
+        public static void ArrayShuffled_NewArrayHaveSameSize(
```
Comment (siriak) [inline Algorithms.Tests/Sorters/Comparison/FisherYatesShufflerTests.cs:11]:
```suggestion
        public static void ArrayShuffled_NewArrayHasSameSize(
```

File: Algorithms.Tests/Sorters/Comparison/FisherYatesShufflerTests.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            // Arrange
+            var shuffler = new FisherYatesShuffler<int>();
+            var (correctArray, testArray) = RandomHelper.GetArrays(n);
+
+            // Act
+            shuffler.Shuffle(testArray);
+
+            // Assert
+            Assert.AreEqual(testArray.Length, correctArray.Length);
```
Comment (siriak) [inline Algorithms.Tests/Sorters/Comparison/FisherYatesShufflerTests.cs:23]: Please use FluentAssertions

**Review by siriak — COMMENTED** [review]
Looks good, could you put it in the Algorithms/Shufflers directory?

**Comment by DPalaSS** [comment]
Updated Fisher-Yates algorithm, please check)

File: Algorithms.Tests/Shufflers/FisherYatesShufflerTests.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
+using Algorithms.Shufflers;
+using Algorithms.Tests.Helpers;
+using FluentAssertions;
+using NUnit.Framework;
+using System;
+
+namespace Algorithms.Tests.Shufflers {
```
Comment (siriak) [inline Algorithms.Tests/Shufflers/FisherYatesShufflerTests.cs:7]:
```suggestion
namespace Algorithms.Tests.Shufflers
{
```

**Review by siriak — APPROVED** [review]
LGTM, thanks!

**Pull request merged**